src/number_theory/kummer_dedekind.lean

```diff
@@ -0,0 +1,328 @@
+/-
+Copyright (c) 2021 Anne Baanen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anne Baanen, Paul Lezeau
+-/
+
+import field_theory.minpoly
+import ring_theory.adjoin_root
+import ring_theory.dedekind_domain.ideal
+import ring_theory.ideal.operations
+import ring_theory.polynomial.basic
+import ring_theory.power_basis
+import ring_theory.unique_factorization_domain
+import data.polynomial.field_division
+import ring_theory.chain_of_divisors
+import algebra.algebra.basic
+
+/-!
+# Kummer-Dedekind theorem
+
+This file proves the Kummer-Dedekind theorem on the splitting of prime ideals in an extension
+of the ring of integers.
+
+## Main definitions
+
+ * `factors_equiv'` : The bijection in the Kummer-Dedekind theorem
+
+## Main results
+
+ * `normalized_factors_ideal_map_eq_normalized_factors_min_poly_mk_map` : The Kummer-Dedekind
+    theorem
+ * `ideal.irreducible_map_of_irreducible_minpoly` : `I.map (algebra_map R S)` is irreducible if
+    `(map I^.quotient.mk (minpoly R pb.gen))` is irreducible, where `pb` is a power basis of `S`
+    over `R`
+
+## TODO
+ * Define the conductor ideal and prove the Kummer-Dedekind theorem in full generality
+ * Prove the converse of `ideal.irreducible_map_of_irreducible_minpoly`
+
+## Tags
+
+kummer, dedekind, kummer dedekind, dedekind-kummer, dedekind kummer
+-/
+
+open_locale big_operators
+
+open ideal polynomial double_quot unique_factorization_monoid
+
+open_locale classical
+
+variables {R : Type*} [comm_ring R] [is_domain R] [is_dedekind_domain R]
+
+variables {S : Type*} [comm_ring S] [algebra R S]
+
+lemma temporary (f : polynomial R) (r : R) :
+  adjoin_root.of f r = adjoin_root.mk f (polynomial.C r) := rfl
+
+open alg_equiv
+
+/-- Let `f` be a polynomial over `R` and `I` an ideal of `R`,
+then `(R[x]/(f)) / (I)` is isomorphic to `(R/I)[x] / (f mod p)` -/
+noncomputable def adjoin_root.quot_equiv_quot_map
+  (f : polynomial R) (I : ideal R) :
+  (_ ⧸ (ideal.map (adjoin_root.of f) I)) ≃ₐ[R]
```

Comment:
The first underscore is just `R[X]`, right? If it is the case I think you can write it explicitly, to make the statement more readable. I am not sure about the second underscore.

In any case this lemma should be in a more basic file. And similarly for some of the lemmas below.